src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -331,247 +145,412 @@ export const FileUpload: React.FC<IFileUploadProps> = ({
     }, []);
 
-    useEffect(() => {
-        // this is to set some values the first time when the component it's expanded
-        if (
-            state.height === undefined &&
-            (isUploading || isSuccess || isFailure)
-        ) {
-            if (containerRef.current?.scrollHeight) {
-                dispatch({
-                    type: SET_TOTAL_HEIGHT_PLUS,
-                    value: containerRef.current.scrollHeight - PADDING * 2,
-                });
-            }
-            if (loadingContainerRef.current?.getBoundingClientRect().top) {
-                dispatch({
-                    type: SET_POSITION_TOP_LOADING,
-                    value:
-                        loadingContainerRef.current.getBoundingClientRect()
-                            .top - MARGIN,
-                });
+    const onWorkerMessage = useCallback(
+        (e: any) => {
+            const { base64StringFile, name } = e.data;
+            if (base64StringFile === undefined) {
+                return;
             }
-        }
-        // this is to calculate and set the width of the success and failure container component
-        if (isSuccess || isFailure) {
-            const width = containerRef.current?.getBoundingClientRect().width;
-            if (width) {
-                dispatch({
-                    type: SET_IS_SUCCESS_WIDTH,
-                    value: width - PADDING * 2 - MARGIN * 2,
-                });
-            }
-        }
-        // this sets height of the component, is used to transition between heights.
-        if (isUploading || isSuccess || isFailure) {
-            if (state.totalHeightPlus) {
-                dispatch({ type: SET_HEIGHT, value: state.totalHeightPlus });
+            if (base64StringFile === NO_BASE64STRINGFILE || isTestIsFailure) {
+                const value = informativePanelsState.values.find(
+                    (value_) => value_.name === name,
+                );
+                if (value) {
+                    value.worker.terminate();
+                    setInformativePanelsState((prev) => ({
+                        ...prev,
+                        values: prev.values.map((value_) => {
+                            if (value_.name === value.name)
+                                return {
+                                    ...value_,
+                                    isSuccess: {
+                                        ...value_.isSuccess,
+                                        value: false,
+                                    },
+                                    isFailure: {
+                                        ...value_.isFailure,
+                                        value: true,
+                                    },
+                                    isUploading: {
+                                        ...value_.isUploading,
+                                        value: false,
+                                    },
```

Comment:
Duplicate code with piece below abstract to function to reduce code duplicate. Use parameters for function to solve issue